polkadot/node/network/gossip-support/src/tests.rs

```diff
@@ -89,17 +95,14 @@ type VirtualOverseer = test_helpers::TestSubsystemContextHandle<GossipSupportMes
 
 #[derive(Debug, Clone)]
 struct MockAuthorityDiscovery {
-	addrs: HashMap<AuthorityDiscoveryId, HashSet<Multiaddr>>,
-	authorities: HashMap<PeerId, HashSet<AuthorityDiscoveryId>>,
+	addrs: Arc<Mutex<HashMap<AuthorityDiscoveryId, HashSet<Multiaddr>>>>,
```

Comment:
could you elaborate why Arc<Mutex<_>> is needed? it's not clear to me by reading code, since `add_more_authorities` takes `&mut self`.

Reply:
Yeah, it doesn't need to be `mut` I don't know why clippy didn't not tell me it is un-ncessary. This needs to be an Arc<Mut> because we are passing a clone of it into the Gossip support system, so if we want to notice changes we do to this mocked object then it needs to be a shallow copy(Arc) and then we need the interior mutability(Mutex) for `add_more_authorties`.